### PR TITLE
Control off-range beta in `anisotropy_parameter()`; avoid broken SciPy optimizer

### PR DESCRIPTION
**Pull request by MikhailRyazanov**
Investigation of the spurious errors and warnings in CI tests (#391) revealed that:
1. New Apple's hardware (arm64) have larger floating-point errors.
2. The default method `'lm'` in `scipy.optimize.curve_fit()` is broken (see SciPy [#21995](https://github.com/scipy/scipy/issues/21995)) and thus was producing `OptimizeWarning` in _some_ testing environments; more importantly, it could produce wrong estimates for fitting errors.

So this PR changes the method in `curve_fit()` to a well-behaved `'trf'` and also adds tests to ensure that the error estimates returned by `anisotropy_parameter()` are correct. Some tolerances in the tests are also increased slightly to accommodate this change and the sloppier Apple's arm64 performance.

Now all the tests pass cleanly.

The `'trf'` method also supports constrained fitting, with is useful for keeping the anisotropy parameter in the physical range. Thus the `anisotropy_parameter()` function is extended here such that, instead of always reporting off-range results as `nan`, it also allows to return them as is or to use a constrained fit. By default, the old behavior (returning `nan`) is preserved.

`radial_integration()` is also extended to accept additional parameters and pass them to `anisotropy_parameter()`.

### CODE REVIEW

**Comment by MikhailRyazanov** [comment]
Suggestions for better names of the `beta_out` parameter (and values) are welcome!

And a question: should we also add `beta_out` (and maybe `theta_ranges`) to `tools.vmi.radial_integration()`, which calls `anisotropy_parameter()` internally?

**Comment by DanHickstein** [comment]
Looks good to me!

`beta_out` seems like a reasonable parameter name. I suppose `beta_behavior` is another option that you can take or leave.

I don't have strong feelings about adding these parameters to `tools.vmi.radial_integration()`.

**Comment by MikhailRyazanov** [comment]
> I suppose `beta_behavior` is another option that you can take or leave.

Or “behavio**u**r”. :–) Maybe then just `beta='raw'/'bound'/???` with a more meaningful name for `'nan'` (which is the default and thus can be cumbersome without causing inconvenience)?

File: abel/tools/vmi.py
Comment (stggh) [inline abel/tools/vmi.py:1]: Not your issue, but `pycodestyle` would prefer a shorter line 200.

**Review by stggh — COMMENTED** [review]
You are very busy @MikhailRyazanov !

> also add beta_out (and maybe theta_ranges) to tools.vmi.radial_integration()

Yes, this has to be done, for consistency

File: abel/tools/vmi.py
Comment (MikhailRyazanov) [reply abel/tools/vmi.py:1]: Where? I see several complaints about other lines (none affected here), but line *200* is empty, and there's nothing long near it...

**Comment by MikhailRyazanov** [comment]
I've renamed this new `beta_out` parameter in `anisotropy_parameter()` to shorter and (I hope) clearer `mode` with options `'raw'`, `'reject'` and `'bound'`.
And also, as agreed, added to `tools.vmi.radial_integration()` the parameters `theta_ranges` and this `mode` to pass directly to `anisotropy_parameter()`.
Please review the documentation for these functions.

Conda is back to life, so now really all the tests pass cleanly everywhere.

File: abel/tools/vmi.py
Comment (stggh) [reply abel/tools/vmi.py:1]: Sorry, that was `test_tools_vmi.py`, so way out of this PR:

```python
(py313)[Gibbo:.+-tmp/abel/tests]$ pycodestyle test_tools_vmi.py 
test_tools_vmi.py:200:80: E501 line too long (83 > 79 characters)
```

File: doc/conf.py [hunk 1]
```diff
@@ -45,8 +45,7 @@
 templates_path = ['_templates']
 
 # The suffix(es) of source filenames.
-# You can specify multiple suffix as a list of string:
-source_suffix = ['.rst']
+#source_suffix = {'.rst': 'restructuredtext'}
```
Comment (stggh) [inline doc/conf.py:48]: Why the change? Is there no need to specify this parameter?  (= for my own education)

**Review by stggh — APPROVED** [review]
Looks fine to me. I like 'mode' for the beta return type parameter, and the clear choices.

Good work @MikhailRyazanov

File: doc/conf.py (hunk 1 above)
Comment (MikhailRyazanov) [reply doc/conf.py:48]: Sphinx was issuing a warning
```
Converting `source_suffix = ['.rst']` to `source_suffix = {'.rst': 'restructuredtext'}`.
```
because the [preferred format for `source_suffix`](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-source_suffix) has actually [changed years ago](https://www.sphinx-doc.org/en/master/changes/1.8.html#release-1-8-0-released-sep-13-2018). But it's also set to `.rst` by default, so there's no need to specify it explicitly, until we want to add something else (there are many lines in `conf.py` that show the default but are commented out). So just for consistency.
